src/Lecture3.hs

```diff
@@ -210,11 +232,26 @@ types that can have such an instance.
   implement instances! No spoilers :)
 -}
 
--- instance Foldable Weekday where
--- instance Foldable Gold where
--- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+-- instance Foldable Weekday where -- Kind is `*`
+-- instance Foldable Gold where -- Kind is `*` 
+-- instance Foldable Reward where -- Kind is `*`
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f b (List1 a []) = f a b
+  foldr f b (List1 a as) = f a (foldr f b as)
```

Comment:
You don't need to handle empty lists explicitly and can remove this case as `foldr` for list already handles it 👍🏻